homeassistant/components/nest/config_flow.py

```diff
@@ -106,6 +108,26 @@ async def async_oauth_create_entry(self, data: dict) -> dict:
         data[DATA_SDM] = {}
         return await super().async_oauth_create_entry(data)
 
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        assert self.is_sdm_api(), "Step only supported for SDM API"
+        persistent_notification.async_dismiss(self.hass, NEST_REAUTH_NOTIFICATION)
+        return await super().async_step_auth(user_input)
+
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        assert self.is_sdm_api(), "Step only supported for SDM API"
+        # This causes the config flow to indicate that it should be reconfigured
+        # by showing a notification and an empty for that takes the user back
+        # to the start of the oauth setup flow.
+        persistent_notification.async_create(
+            self.hass,
+            "Nest needs to be re-authenticated.  Please go to the integrations page to re-configure it.",
+            "Nest re-authentication",
+            NEST_REAUTH_NOTIFICATION,
+        )
+        return self.async_show_form(step_id="user", data_schema=vol.Schema({}))
```

Comment:
Don't move directly to another step's form. Await the step method and put the form in the reauth or a confirm reauth step.

Reply:
What I was trying to do was to not show a form at all.   Jumping to the user step fails because it wants to know the current request, which isn't available in the background thread. However, after thinking about it more it seems simpler to just show the reauth confirm page, which is probably more user friendly anyway.

I've updated this config flow explicitly add a reauth confirmation step.  I also fixed some problems where reauth would create a second config entry since it was not deduping against an entry that already exists.  I added test coverage to verify a few different corner cases.

One problem I still have: Once you reauth and fix the config flow, after the abort step on reauth successful: It doesn't seem to call back to re-setup the entry with the new config.  I need to restart HA for the settings to take effect.  Can you spot what i'm doing wrong?

Thanks for the thorough review.

Reply:
Set up the integration again by reloading the config entry. Use `hass.config_entries.async_reload`.